### PR TITLE
Adding shmem unregister to shutdown function for Micro.

### LINKED ISSUE #234: Error using Micro with shmem when the test finalize
**Description**
Two errors will come up at the end of a test using shmem on Micro:
```
[1571049872.218516000]ERROR: ModuleID=2 Errcode=15 X=1 E=0 T=1
db/DBUrtdb.c:1169/DB_Database_delete_table: table=rt

[1571049872.218609000]ERROR: ModuleID=3 Errcode=3 X=1 E=0 T=1
rt/RTRegistry.c:331/RT_Registry_finalize: dbrc=2

Test ended.
```

**How to Reproduce**
Launch the followings commands for a Perftest-Micro compilation:
Publisher:
```
$ ./bin/x64Linux4gcc7.3.0/debug/perftest_cpp_micro -pub -transport SHMEM -exec 2
```

Subscriber:
```
$ ./bin/x64Linux4gcc7.3.0/debug/perftest_cpp_micro -sub -transport SHMEM
```

**Expected behavior**
Any error should be triggered.

**TODO**:
- [x] Support for C++ Traditional
- [x] First CR [Code Review]()
- [x] Automatic install testing
- [x] Documentation
- [ ] Full [Code Review]()
- [ ] Final Merge into master

### PR DESCRIPTION
**Pull request by alfonso-soto**
This pull request is coming from the issue #234 . When the pull request is merged into "master" the issue will be fixed #234

### CODE REVIEW

File: srcCpp/RTIDDSImpl.cxx [hunk 1]
```diff
@@ -112,6 +112,9 @@ void RTIDDSImpl<T>::Shutdown()
         if (!registry->unregister("wh", NULL, NULL)) {
             //printf("failed to unregister wh\n");
         }
+        if (!registry->unregister("_shmem", NULL, NULL)) {
```
Comment (javmorcas) [inline srcCpp/RTIDDSImpl.cxx]: We need to check if this is > 2.4.11 and only do it for that, right?

Comment (alfonso-soto) [reply srcCpp/RTIDDSImpl.cxx]: I have already tested for 2.4.11. There is no need to make a difference
Any error would be printed.

File: srcDoc/release_notes.rst [hunk 2]
```diff
@@ -54,6 +54,13 @@ What's New in Master
 What's Fixed in Master
 ~~~~~~~~~~~~~~~~~~~~~~
 
+Error using Micro with SHMEM when the test finalize
```
Comment (javmorcas) [inline srcDoc/release_notes.rst]: Error finalizing the application when using `SHMEM` for `RTI Connext DDS Micro` (#234)

File: srcDoc/release_notes.rst [hunk 3]
```diff
@@ -54,6 +54,13 @@ What's New in Master
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+An error using *RTI Perftest* for micro 3.0.0 has been solved by unregistering
+'_shmem' before finalize_instance().
```
Comment (javmorcas) [inline srcDoc/release_notes.rst]: When using *RTI Connext DDS Micro* and setting the transport to `SHMEM` an error would appear at the end of the test in both publisher and subscriber by the time he `finalize_instance()` function is called. This erros has been resolved.

File: srcCpp/RTIDDSImpl.cxx (hunk 1 above)
Comment (javmorcas) [inline srcCpp/RTIDDSImpl.cxx]: add a comment specifying that this will only be successful for Micro 3.0.0 and above.

Comment (alfonso-soto) [reply srcCpp/RTIDDSImpl.cxx]: Done

**Pull request merged**